Mathlib/Order/FunLike.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2024 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Data.FunLike.Basic
+import Mathlib.Order.Basic
+
+/-!
+# Preorder and partial order on bundled homs
+
+In this file we define `LE`, `Preorder`, and `PartialOrder` instances
+on any type with `DFunLike` instance and appropriate instances on the codomains.
+-/
+
+namespace DFunLike
+
+variable {F : Type*} {α : Type*} {β : α → Type*} [DFunLike F α β]
+
+section LE
+
+variable (F) in
+/-- A mixin typeclass saying that the order on bundled morphisms
+agrees with the pointwise order on corresponding functions. -/
```

Comment:
```suggestion
agrees with the pointwise order on corresponding functions.

Warning: Using this typeclass will set `f < g` to `f ≤ g ∧ ¬ g ≤ f`.
Make sure that this is defeq to your definition of `<` to avoid diamonds. -/
```

Reply:
**This** typeclass says nothing about `f < g`.

Reply:
Oh right, because `instPartialOrder` has [Preorder F]`